airflow/dag_processing/bundles/base.py

```diff
@@ -112,3 +125,22 @@ def view_url(self, version: str | None = None) -> str | None:
         :param version: Version to view
         :return: URL to view the bundle
         """
+
+    @contextmanager
+    def lock(self):
+        if self._locked:
+            yield
+            return
+
+        lock_dir_path = self._dag_bundle_root_storage_path / "locks"
```

Comment:
maybe `_locks`? to avoid having to "reserve" the word?

Reply:
Yeah, fair! I'll do it after this test run.